src/python/zquantum/optimizers/basin_hopping.py

```diff
@@ -0,0 +1,96 @@
+from zquantum.core.interfaces.optimizer import Optimizer, optimization_result
+from zquantum.core.history.recorder import recorder
+from typing import Callable, Dict, Union
+import scipy.optimize
+
+
+class BasinHoppingOptimizer(Optimizer):
+    def __init__(
+        self,
+        keep_value_history: bool = False,
+        niter: int = 100,
+        T: float = 1.0,
+        stepsize: float = 0.5,
+        minimizer_kwargs: Union[dict, None] = None,
+        take_step: Union[Callable, None] = None,
+        accept_test: Union[Callable, None] = None,
+        interval: int = 50,
+        disp: bool = False,
+        niter_success: Union[int, None] = None,
+    ):
+        """
+        Args:
+            keep_value_history: determines whether or not the cost function records when it is evaluated
+            niter: See scipy.optimize.basinhopping
```

Comment:
Since all the arguments are described in the scipy docs, perhaps a link there would be appropriate?